codemod/base.py

```diff
@@ -0,0 +1,1000 @@
+                        help='A path:line_number-formatted position somewhere'
+                             ' in the hierarchy from which to being exploring,'
+                             'or a percentage (e.g. "--start 25%%") of '
+                             'the way through to start.'
+                             'Useful if you\'re divvying up the '
+                             'substitution task across multiple people.')
+    parser.add_argument('--end', action='store', type=str,
+                        help='A path:line_number-formatted position '
+                             'somewhere in the hierarchy just *before* '
+                             'which we should stop exploring, '
+                             'or a percentage of the way through, '
+                             'just before which to end.')
+
+    parser.add_argument('--extensions', action='store', type=str,
+                        help='A comma-delimited list of file extensions '
+                             'to process.')
+    parser.add_argument('--include-extensionless', action='store_true',
+                        help='If set, this will check files without '
+                        'an extension, along with any matching file '
+                        'extensions passed in --extensions')
+    parser.add_argument('--exclude-paths', action='store', type=str,
+                        help='A comma-delimited list of paths to exclude.')
+
+    parser.add_argument('--accept-all', action='store_true',
+                        help='Automatically accept all '
+                             'changes (use with caution).')
+
+    parser.add_argument('--default-no', action='store_true',
+                        help='If set, this will make the default '
+                             'option to not accept the change.')
+
+    parser.add_argument('--editor', action='store', type=str,
+                        help='Specify an editor, e.g. "vim" or emacs". '
+                        'If omitted, defaults to $EDITOR environment '
+                        'variable.')
+    parser.add_argument('--count', action='store_true',
+                        help='Don\'t run normally.  Instead, just print '
+                             'out number of times places in the codebase '
+                             'where the \'query\' matches.')
+    parser.add_argument('--test', action='store_true',
+                        help='Don\'t run normally.  Instead, just run '
+                             'the unit tests embedded in the codemod library.')
+
+    parser.add_argument('match', nargs='?', action='store', type=str,
+                        help='Regular expression to match.')
+    parser.add_argument('subst', nargs='?', action='store', type=str,
+                        help='Substitution to replace with.')
+
+    arguments = parser.parse_args()
+
+    if arguments.test:
+        import doctest
+        doctest.testmod()
+        sys.exit(0)
```

Comment:
Since this pull request introduces `pytest` as a method of discovering doc tests, shouldn't this be removed? Feel free to add a commit to the stack in this pull request, or just open an issue on the repo so we can do it later.